diff --git a/kstyle/breezemetrics.h b/kstyle/breezemetrics.h
index 9425b07..e9a7d7b 100644
--- a/kstyle/breezemetrics.h
+++ b/kstyle/breezemetrics.h
@@ -90,7 +90,7 @@ namespace Breeze
 
         // scrollbars
-        static constexpr int ScrollBar_Extend = 21;
+        static constexpr int ScrollBar_Extend = 16;
-        static constexpr int ScrollBar_SliderWidth = 10;
+        static constexpr int ScrollBar_SliderWidth = 6;
         static constexpr int ScrollBar_MinSliderHeight = 20;
         static constexpr int ScrollBar_NoButtonHeight = (ScrollBar_Extend-ScrollBar_SliderWidth)/2;
         static constexpr int ScrollBar_SingleButtonHeight = ScrollBar_Extend;
