diff --git a/kconf_update/gtktheme.cpp b/kconf_update/gtktheme.cpp
index a18f2f6..a4b387a 100644
--- a/kconf_update/gtktheme.cpp
+++ b/kconf_update/gtktheme.cpp
@@ -48,7 +48,7 @@ void upgradeGtk3Theme()
 
 QString gtk2Theme()
 {
-    QString gtkrcPath = QDir::homePath() + QStringLiteral("/.gtkrc-2.0");
+    QString gtkrcPath = QDir::homePath() + QStringLiteral("/.config/gtk-2.0/gtkrc");
     QFile gtkrc(gtkrcPath);
     if (gtkrc.open(QIODevice::ReadWrite | QIODevice::Text)) {
         const QRegularExpression regExp(QStringLiteral("gtk-theme-name=[^\n]*($|\n)"));
diff --git a/kded/configeditor.cpp b/kded/configeditor.cpp
index c3476b7..6fa13ee 100644
--- a/kded/configeditor.cpp
+++ b/kded/configeditor.cpp
@@ -105,9 +105,9 @@ void ConfigEditor::setGtk3ConfigValueXSettingsd(const QString &paramName, const
 
 void ConfigEditor::setGtk2ConfigValue(const QString &paramName, const QVariant &paramValue)
 {
-    QString gtkrcPath = qEnvironmentVariable("GTK2_RC_FILES", QDir::homePath() + QStringLiteral("/.gtkrc-2.0"));
+    QString gtkrcPath = qEnvironmentVariable("GTK2_RC_FILES", QDir::homePath() + QStringLiteral("/.config/gtk-2.0/gtkrc"));
     if (gtkrcPath.contains(QStringLiteral(":/"))) { // I.e. env variable contains multiple paths
-        gtkrcPath = QDir::homePath() + QStringLiteral("/.gtkrc-2.0");
+        gtkrcPath = QDir::homePath() + QStringLiteral("/.config/gtk-2.0/gtkrc");
     }
     QFile gtkrc(gtkrcPath);
     QString gtkrcContents = readFileContents(gtkrc);
@@ -138,7 +138,7 @@ void ConfigEditor::setGtk3Colors(const QMap<QString, QColor> &colorsDefinitions)
 
 void ConfigEditor::removeLegacyGtk2Strings()
 {
-    QString gtkrcPath = QDir::homePath() + QStringLiteral("/.gtkrc-2.0");
+    QString gtkrcPath = QDir::homePath() + QStringLiteral("/.config/gtk-2.0/gtkrc");
     QFile gtkrc(gtkrcPath);
     QString gtkrcContents = readFileContents(gtkrc);
 
