--- a/kstyle/breezemetrics.h
+++ b/kstyle/breezemetrics.h
@@ -89,8 +89,8 @@ namespace Breeze
         static constexpr int MenuBarItem_MarginHeight = 6;
 
         // scrollbars
-        static constexpr int ScrollBar_Extend = 21;
-        static constexpr int ScrollBar_SliderWidth = 8;
+        static constexpr int ScrollBar_Extend = 16;
+        static constexpr int ScrollBar_SliderWidth = 6;
         static constexpr int ScrollBar_MinSliderHeight = 20;
         static constexpr int ScrollBar_NoButtonHeight = (ScrollBar_Extend-ScrollBar_SliderWidth)/2;
         static constexpr int ScrollBar_SingleButtonHeight = ScrollBar_Extend;
