diff --git a/libaegisub/unix/path.cpp b/libaegisub/unix/path.cpp
index 0541e0d..735691c 100644
--- a/libaegisub/unix/path.cpp
+++ b/libaegisub/unix/path.cpp
@@ -35,6 +35,30 @@ std::string home_dir() {
 
 	throw agi::EnvironmentError("Could not get home directory. Make sure HOME is set.");
 }
+
+std::string xdg_config_home_dir() {
+	const char *env = getenv("XDG_CONFIG_HOME");
+	if (env && *env) return env;
+
+	agi::fs::path home = home_dir();
+	return (home/".config").string();
+}
+
+std::string xdg_cache_home_dir() {
+	const char *env = getenv("XDG_CACHE_HOME");
+	if (env && *env) return env;
+
+	agi::fs::path home = home_dir();
+	return (home/".cache").string();
+}
+
+std::string xdg_data_home_dir() {
+	const char *env = getenv("XDG_DATA_HOME");
+	if (env && *env) return env;
+
+	agi::fs::path home = home_dir();
+	return (home/".local/share").string();
+}
 #endif
 }
 
@@ -42,8 +66,17 @@ namespace agi {
 void Path::FillPlatformSpecificPaths() {
 #ifndef __APPLE__
 	agi::fs::path home = home_dir();
-	SetToken("?user", home/".aegisub");
-	SetToken("?local", home/".aegisub");
+    agi::fs::path prev_dir = home/".aegisub";
+	if (!boost::filesystem::exists(prev_dir))
+	{
+		agi::fs::path xdg_config_home = xdg_config_home_dir();
+		agi::fs::path xdg_cache_home = xdg_cache_home_dir();
+		SetToken("?user", xdg_config_home/"aegisub");
+		SetToken("?local", xdg_cache_home/"aegisub");
+	} else {
+		SetToken("?user", prev_dir);
+	    SetToken("?local", prev_dir);
+	}
 	SetToken("?data", P_DATA);
 	SetToken("?dictionary", "/usr/share/hunspell");
 #else
